### ISSUE TITLE
More benchmarks

### ISSUE THREAD

**Issue by tju01** [tju01/ilm-eval#1]
Right now, this project only includes evaluations for OpenAssistant on OpenAI evals. This results in a limited number and especially type of tasks. It would be good to also use other benchmarks and combine the resulting scores with the scores from OpenAI evals. This should then also allow for distinguishing the quality of models that are closer together.

There are basically two types of benchmarks for LLMs. First, we have benchmarks that are for pre-trained LLMs that have not been fine-tuned. These benchmarks work by few-shot prompting (giving some Q/A examples and then asking for the last A for the last Q). Second, we have benchmarks that are specifically for fine-tuned models. They work with zero-shot prompting, i.e. directly asking the question.

### Benchmarks for fine-tuned models

For OpenAssistant, we are mostly interested in benchmarks for fine-tuned models, because that's what OpenAssistant is about. Unfortunately, that excludes the vast majority of benchmarks. I have only found the following benchmarks that seem to be specifically for automated evaluation of fine-tuned models:
- [OpenAI evals](https://github.com/openai/evals/), which is what this repository is currently about.
- [Vicuna-style evaluation](https://lmsys.org/blog/2023-03-30-vicuna/) which uses another language model like GPT-4 for evaluation.

### Benchmarks for pre-trained models

Apart from these benchmarks for fine-tuned models, there are tons of benchmarks for pre-trained models. It doesn't really make sense to add all of them separately. Instead, one should probably use [HELM](https://crfm.stanford.edu/helm/) or [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness) for a unified interface to a large collection of benchmarks.

While OpenAssistant is about fine-tuning, these benchmarks still play a role. This is due to the following two reasons:
- We want to ensure that fine-tuning doesn't make the model forget too much knowledge. So comparing the fine-tuned model to the pre-trained model and other fine-tuned models makes sense in order to quantify knowledge loss from fine-tuning.
- In the [HELM paper](https://arxiv.org/abs/2211.09110), they actually talk about "The benefits of instruction-tuning". It can actually improve results even on those benchmarks apparently. Haven't read it too deeply though, needs more research.

**Comment by tju01** [comment]
I'm working on the [Vicuna-style evaluation](https://lmsys.org/blog/2023-03-30-vicuna/) now. See the `fastchat-eval` folder.

**Comment by tju01** [comment]
Added some initial version of the vicuna benchmark. Going to work on lm-evaluation-harness next.

**Comment by tju01** [comment]
Added lm-evaluation-harness.
